### PR TITLE
Add/block preview

### LINKED ISSUE #26289: Allow customization of the viewportWidth for the BlockPreview
## Use case

Allow the viewportWidth to be customized by a developer who register a block using the function register_block_type.
This would improve the way the blocks is previewed with the best possible render.

20201019-mberard-screenshot

## Suggested Solution

Instead of having a fixed value for the props viewportWidth, allow it to be override by developer using example object. For instance, with a .example.viewportWidth property.

https://github.com/WordPress/gutenberg/blob/a5698b03dd2ccc787e8734b6e6b20b464a78fd7f/packages/block-editor/src/components/inserter/preview-panel.js#L27

### PR DESCRIPTION
**Pull request by jeff-be**
## Description

closes https://github.com/WordPress/gutenberg/issues/26289

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: docs/designers-developers/developers/block-api/block-registration.md [hunk 1]
```diff
@@ -204,11 +204,22 @@ example: {
                 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent et eros eu felis.'
             ),
         },
-    },
+	},
```
Comment (youknowriad) [inline docs/designers-developers/developers/block-api/block-registration.md:207]: should we indent the whole object here using tabs? (otherwise, we might want to keep this for a separate PR and revert this line)

File: packages/block-editor/src/components/block-preview/README.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ Width of the preview container in pixels. Controls at what size the blocks will
 
 `viewportWidth` can be used to simulate how blocks look on different device sizes or to make sure make sure multiple previews will be rendered with the same scale, regardless of their content.
 
+`viewportWidth` can be override in example object with the property `viewportWidth`.
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-preview/README.md:33]: BlockPreview component can be used in very different contexts, I don't think we should be mentioning the "example" object here. Can we leave this out?

File: packages/block-editor/src/components/inserter/preview-panel.js [hunk 3]
```diff
@@ -24,7 +24,12 @@ function InserterPreviewPanel( { item } ) {
 					<div className="block-editor-inserter__preview-content">
 						<BlockPreview
 							__experimentalPadding={ 16 }
-							viewportWidth={ 500 }
+							viewportWidth={
+								( hoveredItemBlockType.example &&
+									hoveredItemBlockType.example
+										.viewportWidth ) ||
+								500
```
Comment (youknowriad) [inline packages/block-editor/src/components/inserter/preview-panel.js:31]: Potentially you could have written `hoveredItemBlockType.example?.viewportWidth ?? 500`

**Review by youknowriad — COMMENTED** [review]
Thanks for the PR this looks like a nice API.

You could have expanded a bit more on the PR description to help reviewers understand and test the PR.

File: docs/designers-developers/developers/block-api/block-registration.md (hunk 1 above)
Comment (jeff-be) [reply docs/designers-developers/developers/block-api/block-registration.md]: Done

File: packages/block-editor/src/components/block-preview/README.md (hunk 2 above)
Comment (jeff-be) [reply packages/block-editor/src/components/block-preview/README.md]: Done

File: packages/block-editor/src/components/inserter/preview-panel.js (hunk 3 above)
Comment (jeff-be) [reply packages/block-editor/src/components/inserter/preview-panel.js:29]: Done

**Comment by jeff-be** [comment]
Thanks for review.

Sorry for the lack of description, it's my first PR. But I take note for the next one

File: packages/block-editor/src/components/block-preview/README.md (hunk 2 above)
Comment (youknowriad) [reply packages/block-editor/src/components/block-preview/README.md]: Oh sorry, I was not clear enough, I was thinking we should remove the whole sentence from here.

Comment (jeff-be) [reply packages/block-editor/src/components/block-preview/README.md]: OK it's done

**Review by youknowriad — APPROVED** [review]
This is looking good. Thanks for the addition.
